tests/tamr_client/fake_json/dataset/test_dataset/test_by_name.json

```diff
@@ -0,0 +1,36 @@
+[
+    {
+        "request": {
+            "method": "GET",
+            "path": "datasets?filter=name==dataset 1 name"
```

Comment:
nit: Ideally, tests should be as realistic as possible, so `dataset 1 name` should be replaced with something like `Customers_2019.csv` or something. We are inconsistent about this and I think we can wait to fix this until when we switch over to faking a test server.